### PR TITLE
Update/cleanup RAJAPerf Caliper integration

### PR DESCRIPTION
**Pull request by daboehme**
Some cleanup in the RAJAPerf Caliper integration:

- Move flush to `outputRunData()`
- Move implementations from KernelBase.hpp into KernelBase.cpp
- Simplify option spec for the extra attributes

### CODE REVIEW

File: src/rajaperf_config.hpp.in [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include "RAJA/config.hpp"
 #include "camp/number.hpp"
 
+#include <algorithm>
```
Comment (MichaelMcKinsey1) [inline src/rajaperf_config.hpp.in:26]: Move this under line 42 since it's only used for caliper?

Comment (daboehme) [reply src/rajaperf_config.hpp.in:26]: Yeah that would be cleaner

Comment (MichaelMcKinsey1) [reply src/rajaperf_config.hpp.in:26]: Can you move it? I don't think I can modify your branch.

Comment (daboehme) [reply src/rajaperf_config.hpp.in]: > Can you move it? I don't think I can modify your branch.

Ok, done.

**Pull request merged**